circuits/core/src/bhp/hash.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+use snarkvm_curves::{MontgomeryParameters, TwistedEdwardsParameters};
+
+impl<E: Environment, const NUM_WINDOWS: usize, const WINDOW_SIZE: usize> BHPCRH<E, NUM_WINDOWS, WINDOW_SIZE> {
+    pub fn hash(&self, input: &[Boolean<E>]) -> Field<E> {
+        self.hash_bits_inner(input).to_x_coordinate()
+    }
+
+    fn hash_bits_inner(&self, input: &[Boolean<E>]) -> Group<E> {
+        // Ensure the input size is within the parameter size,
+        if input.len() > NUM_WINDOWS * WINDOW_SIZE * BHP_CHUNK_SIZE {
+            E::halt("Incorrect input length")
+        }
+
+        // Pad the input to a multiple of `BHP_CHUNK_SIZE` for hashing.
+        let mut input = input.to_vec();
+        if input.len() % BHP_CHUNK_SIZE != 0 {
+            let padding = BHP_CHUNK_SIZE - (input.len() % BHP_CHUNK_SIZE);
+            input.extend_from_slice(&vec![Boolean::constant(false); BHP_CHUNK_SIZE][..padding]);
```

Comment:
```suggestion
            input.extend_from_slice(&[Boolean::constant(false); BHP_CHUNK_SIZE][..padding]);
```

Reply:
@ljedrz the compiler doesn't seem to like the suggestion. Any alternative options?

Reply:
What does the compiler complain about? I just double-checked it locally and it works.

Reply:
It said it doesn't implement `Copy`

Reply:
Okay strange, I confirm it works locally on my machine too. It does not work on CI with the `Copy` error.

Reply:
I was able work around it, I'm not sure what went wrong, but cleared cache and started working again. Perhaps CircleCI had a hiccup